cmd/restic/lock.go

```diff
@@ -32,7 +32,39 @@ func lockRepository(ctx context.Context, repo *repository.Repository, exclusive
 		lockFn = restic.NewExclusiveLock
 	}
 
-	lock, err := lockFn(ctx, repo)
+	retrySleep := 10 * time.Second
+	if retrySleep > globalOptions.WaitLock/4 {
+		// Reduce the time between the retries if the maximum total wait
+		// duration is small so that there are at least a few attempts.
+		retrySleep = globalOptions.WaitLock / 4
```

Comment:
For short intervals one retry should be enough. My suggestion is to change this to: `globalOptions.WaitLock*2/3`.  Or just leave the retry interval as is and always add a final retry to the loop.